tests/integration/targets/ndo_schema_template_contract_service_chain/tasks/main.yml

```diff
@@ -0,0 +1,988 @@
+  cisco.mso.ndo_schema_template_contract_service_chain: &cm_add_service_chain
+    <<: *mso_info
+    schema: ansible_test
+    template: ansible_template
+    contract: Contract1
+    node_filter: "allow_all"
+    service_nodes:
+      - device_type: load_balancer
+        consumer_interface_name: lb_interface1
+        provider_interface_name: lb_interface2
+        device:
+          name: load_balancer
+          template: ansible_device_template
+    state: present
+  check_mode: true
+  register: cm_add_service_chain
+
+- name: Create contract service chain - $nm_add_service_chain
+  cisco.mso.ndo_schema_template_contract_service_chain:
+    <<: *cm_add_service_chain
+  register: nm_add_service_chain
+
+- name: Create contract service chain again - $nm_add_service_chain_again
+  cisco.mso.ndo_schema_template_contract_service_chain:
+    <<: *cm_add_service_chain
+  register: nm_add_service_chain_again
+
+- name: Assertion check for create service chain
+  ansible.builtin.assert:
+    that:
+      - cm_add_service_chain is changed
+      - cm_add_service_chain.current.nodeFilter == "allow-all"
+      - cm_add_service_chain.current.serviceNodes.0.consumerConnector.interfaceName == "lb_interface1"
+      - cm_add_service_chain.current.serviceNodes.0.consumerConnector.isRedirect is not defined
+      - cm_add_service_chain.current.serviceNodes.0.deviceName == "load_balancer"
+      - cm_add_service_chain.current.serviceNodes.0.deviceRef != ""
+      - cm_add_service_chain.current.serviceNodes.0.deviceTemplateId != ""
+      - cm_add_service_chain.current.serviceNodes.0.deviceTemplateName == "ansible_device_template"
+      - cm_add_service_chain.current.serviceNodes.0.deviceType == "loadBalancer"
+      - cm_add_service_chain.current.serviceNodes.0.index == 1
+      - cm_add_service_chain.current.serviceNodes.0.name == "node-1"
+      - cm_add_service_chain.current.serviceNodes.0.providerConnector.interfaceName == "lb_interface2"
+      - cm_add_service_chain.current.serviceNodes.0.providerConnector.isRedirect is not defined
+      - cm_add_service_chain.previous == {}
+      - cm_add_service_chain.proposed.nodeFilter == "allow-all"
+      - cm_add_service_chain.proposed.serviceNodes.0.consumerConnector.interfaceName == "lb_interface1"
+      - cm_add_service_chain.proposed.serviceNodes.0.consumerConnector.isRedirect is not defined
+      - cm_add_service_chain.proposed.serviceNodes.0.deviceRef != ""
+      - cm_add_service_chain.proposed.serviceNodes.0.deviceType == "loadBalancer"
+      - cm_add_service_chain.proposed.serviceNodes.0.index == 1
+      - cm_add_service_chain.proposed.serviceNodes.0.name == "node-1"
+      - cm_add_service_chain.proposed.serviceNodes.0.providerConnector.interfaceName == "lb_interface2"
+      - cm_add_service_chain.proposed.serviceNodes.0.providerConnector.isRedirect is not defined
+      - cm_add_service_chain.current.schemaId is defined
+      - cm_add_service_chain.current.schemaName == "ansible_test"
```

Comment:
is templateId and templateName something that should also be there?

Reply:
Done!